src/operations/update.ts

```diff
@@ -23,8 +24,11 @@ export interface UpdateOptions extends CommandOperationOptions {
   let?: Document;
 }
 
-/** @public */
-export interface UpdateResult {
+/**
+ * @public
+ * `TSchema` is the schema of the collection
+ */
+export interface UpdateResult<TSchema = Document> {
```

Comment:
```suggestion
export interface UpdateResult<TSchema extends Document = Document> {
```

We should have the same constraint here that `Collection` has so that this type doesn't support something more than it would via the `Collection<TSchema>` generic.